### ISSUE TITLE
[BUG] False Alarm on Model Training Memory

### ISSUE THREAD

**Issue by FailedPipeline** [autogluon/autogluon#3405]
Training LightGBM model 5 folds, 3 sets on autogluon 0.8.2 and 0.6.2 trips into sequential fitting on a machine with more than enough memory

`	Available Memory:                    612938.11 MB
	Train Data (Original)  Memory Usage: 4202.01 MB (0.7% of available memory)
`

yet ag throws:
`minimum_model_resources: {'num_cpus': 1}
user_cpu_per_job, user_gpu_per_job None | None
user_ensemble_cpu, user_ensemble_gpu None | None
Resources info for CpuResourceCalculator: {'resources_per_job': {'cpu': 4}, 'num_parallel_jobs': 15, 'batches': 1, 'cpu_per_job': 4}
Upper level total_num_cpus, num_gpus 64 | 0
	Dropped 0 of 167 features.
minimum_model_resources: {'num_cpus': 1}
user_cpu_per_job, user_gpu_per_job None | None
user_ensemble_cpu, user_ensemble_gpu None | None
	Memory not enough to fit LGBModel folds in parallel. Will do sequential fitting instead. 	Consider decreasing folds trained in parallel by passing num_folds_parallel to ag_args_ensemble when calling predictor.fit
	Fitting 15 child models (S1F1 - S3F5) | Fitting with SequentialLocalFoldFittingStrategy`

This occurs even though I attempt to force parallel with:
`        predictor = TabularPredictor(label=label, path=ag_model_save_path) \
            .fit(train_data, num_bag_folds=bag_folds, num_bag_sets=bag_sets, num_stack_levels=stack_levels,
                 excluded_model_types=excluded_model_types, feature_prune_kwargs=prune_dict, verbosity=verbosity_level,
                 ag_args_ensemble={'fold_fitting_strategy': 'parallel_local'},
                 hyperparameters={'GBM': ['GBMLarge']})`

This model should easily be able to fit parallel on this machine. Something is being overly squeamish about memory headroom it seems.

Thank you.
